bdk-ffi/src/lib.rs

```diff
@@ -79,6 +79,7 @@ use bitcoin_ffi::FeeRate;
 use bitcoin_ffi::Network;
 use bitcoin_ffi::OutPoint;
 use bitcoin_ffi::Script;
+use bitcoin_ffi::Txid;
```

Comment:
I don't particularly like the `Txid` type from bitcoin-ffi and think we should not use it. See https://github.com/bitcoindevkit/bitcoin-ffi/issues/20 for details. In this case here it's a return value from the Rust side so we know it will be correct, but even so, I think using the type in its current form is not a good idea. Open to hear other people's thoughts on this, but IMO we should stick to `String` for txids for now.

Reply:
Using the type allows for a more robust API in terms of traits. I directly export `NodeEventHandler` straight from `bdk_kyoto` in my PR which uses the actual `Txid`, but it gets coerced to a string for UniFFI users. I find that particularly nice, because Rust design choices have no effect on bindings in that scenario. 

`ldk-node` basically uses string type aliases for almost every struct and event (i.e. payment invoice, preimage, channel ID). I think the correct approach is to use the abstraction if we are confident it is correct.

Reply:
My beef with it is the uncatchable, (runtime!) panic of your whole app if you or your users attempt for whatever reason to build one of those types incorrectly (a txid missing a letter at the end? The whole app collapses and you can't prevent it!).

@tnull how do you see this? Do you mitigate for this in any way? It just feels like a super dangerous type to have for bindings, and is the opposite of type-safety, because the type does _not_ behave the way you'd want/expect it to, but the fact that it exists implies type-safety, whereas if you have a String it's clear that it's not type-safe, and you have at least that small setting of expectations.

To be clear I'm not against a Txid type at all, I just would like something that can throw an error if you try to build an incorrect one.

Reply:
FYI: 

Moving to `string` for this PR. Then when we decide *if* we want to go with `Txid` we can have a follow up PR to change all instances of txid `string` to `Txid` type across the codebase.

Discussion can definitely continue here.

PR is ready to go for merge though.

Reply:
> you or your users attempt for whatever reason to build one of those types incorrectly

There is a big different here between using `Txid` as a return value and an argument. I could see the argument is strong for user-supplied values from potentially untrusted sources. Server provides "" for a response? Unlucky. However for return values that `bdk-ffi` constructs, are we sure the user can even mutate these? It seems abundantly safe to me to return `Txid` as opposed to `string` if we expect the user will treat it as a `Txid`, _especially_ if they aren't allow to mutate `Txid`

Reply:
I agree with the fact that since it is returned, those specific txids it will not cause problems for the users. The problem (as I imagine it at least) is that the type now exists freely at the library level and users can create objects of that type themselves; it's implied that we believe in and support this type as representative of a `Txid` (maybe with the added complication that users familiar with the rust-bitcoin library will think it's a great, well thought-out type when really it's just a string but I'm willing to disregard this for a moment).

Devs wanting to write typesafe software are likely to just start using it in other places (maybe in data structures they fully own), and the type will work more-or-less for them (they can build a txid `abcd` and nothing will get thrown). This is _a little bad_ (because they think they're being typesafe but there are no checks for this Txid being valid in any way), but not _terrible_ (as long as they don't try to use it in a bdk method their apps will not crash and burn). Over time however, if we widen our use of the type and start using it as an argument in methods/constructors, then... kaboom.